### PR TITLE
Default to tick-based updates

### LINKED ISSUE #1444: remove polling watcher
notify based PollWatcher does a ton of caching and can be expensive in giant repos. the new `--polling` arg should sidestep using notify entirely and go back to a regular status polling

### PR DESCRIPTION
**Pull request by cruessler**
This Pull Request fixes/closes #1444.

It changes the following:

- This commit reintroduces code that was previously removed in favor of a notify-based update trigger. It turned out that notify-based updates can cause issues in larger repositories, so tick-based updates seemed like a safer default.

PR that introduced the notify-based watcher: https://github.com/extrawurst/gitui/pull/1310

I followed the checklist:

- [ ] I added unittests
- [x] I ran `make check` without errors
- [x] I tested the overall application
- [x] I added an appropriate item to the changelog

### CODE REVIEW

File: src/args.rs [hunk 1]
```diff
@@ -15,6 +15,7 @@ use std::{
 pub struct CliArgs {
 	pub theme: PathBuf,
 	pub repo_path: RepoPath,
+	pub notify_watcher: bool,
 	pub poll_watcher: bool,
```
Comment (extrawurst) [inline src/args.rs:19]: `poll_watcher` should be deprecated by this, right? because it is the default

File: src/args.rs [hunk 2]
```diff
@@ -95,6 +99,12 @@ fn app() -> ClapApp {
 				.long("logging")
 				.num_args(0),
 		)
+		.arg(
+			Arg::new("watcher")
+				.help("Use notify-based file system watcher instead of tick-based update. This is more performant, but can cause issues in larger repositories")
```
Comment (extrawurst) [inline src/args.rs:104]: maybe we can add a entry in `FAQ` to sum up the reasonings behind this and link to the FAQ here

**Review by extrawurst — COMMENTED** [review]
Thanks for taking this one ❤️

File: src/main.rs [hunk 3]
```diff
@@ -83,11 +83,13 @@ use tui::{
 use ui::style::Theme;
 use watcher::RepoWatcher;
 
+static TICK_INTERVAL: Duration = Duration::from_secs(5);
```
[the remark this answers is not in the document]
Comment (cruessler) [reply src/main.rs:86]: Yes, I got this line from #1310.

File: src/main.rs [hunk 4]
```diff
@@ -114,6 +116,13 @@ pub enum AsyncNotification {
 	Git(AsyncGitNotification),
 }
 
+#[derive(Clone, Copy, PartialEq)]
+enum Updater {
+	Ticker,
+	PollWatcher,
```
[the remark this answers is not in the document]
Comment (cruessler) [reply src/main.rs:122]: No, if it’s causing issues that are not fixable or unlikely to be fixed, it makes more sense to completely remove it.

File: src/args.rs (hunk 2 above)
Comment (cruessler) [reply src/args.rs]: I added an entry to `FAQ.md` and referenced it here.

File: src/args.rs (hunk 1 above)
Comment (cruessler) [reply src/args.rs]: I followed your suggestion below and removed `poll_watcher`.

After the removal, `clap` prints an error when `--poll` is passed as an argument: `error: unexpected argument '--poll' found`.

**Pull request merged**